target/linux/mediatek/dts/mt7981b-totolink-x6000r.dts

```diff
@@ -0,0 +1,279 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+
+/dts-v1/;
+
+#include <dt-bindings/leds/common.h>
+
+#include "mt7981b.dtsi"
+
+/ {
+	model = "TOTOLINK X6000R";
+	compatible = "totolink,x6000r", "mediatek,mt7981";
+
+	aliases {
+		label-mac-device = &gmac0;
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+		serial0 = &uart0;
+	};
+
+    chosen {
+        stdout-path = "serial0:115200n8";
+    };
+
+    gpio-keys {
+        compatible = "gpio-keys";
+
+        reset {
+            label = "reset";
+            linux,code = <KEY_RESTART>;
+            gpios = <&pio 0 GPIO_ACTIVE_LOW>;
+        };
+
+        wps {
+            label = "wps";
+            linux,code = <KEY_WPS_BUTTON>;
+            gpios = <&pio 1 GPIO_ACTIVE_LOW>;
+        };
+    };
+
+    gpio-leds {
+        compatible = "gpio-leds";
+
+        led-0 {
+            function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_GREEN>;
+            gpios = <&pio 8 GPIO_ACTIVE_LOW>;
+        };
+
+        led-1 {
+			color = <LED_COLOR_ID_RED>;
+            gpios = <&pio 13 GPIO_ACTIVE_LOW>;
+        };
+
+        led_status: led-3 {
+			color = <LED_COLOR_ID_BLUE>;
+            function = LED_FUNCTION_STATUS;
+            gpios = <&pio 15 GPIO_ACTIVE_LOW>;
+            default-state = "on";
+        };
+    };
+};
+
+&uart0 {
+    status = "okay";
+};
+
+&eth {
+	pinctrl-names = "default";
+	pinctrl-0 = <&mdio_pins>;
+
+	status = "okay";
+
+	gmac0: mac@0 {
+		compatible = "mediatek,eth-mac";
+		reg = <0>;
+		phy-mode = "2500base-x";
+
+		nvmem-cells = <&macaddr_factory_e006>;
+		nvmem-cell-names = "mac-address";
+		fixed-link {
+			speed = <2500>;
+			full-duplex;
+			pause;
+		};
+	};
+
+	gmac1: mac@1 {
+		compatible = "mediatek,eth-mac";
+		reg = <1>;
+		phy-mode = "gmii";
+		nvmem-cells = <&macaddr_factory_e01a>;
+		nvmem-cell-names = "mac-address";
+	};
+
+	mdio: mdio-bus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+	};
+};
+
+&mdio_bus {
+	switch: switch@1f {
+		compatible = "mediatek,mt7531";
+		reg = <31>;
+		reset-gpios = <&pio 39 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&spi0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi0_flash_pins>;
+	status = "disabled";
+};
+
+&spi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi2_flash_pins>;
+	status = "okay";
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+
+		spi-max-frequency = <48000000>;

```

Comment:
The MT7981 supports an SPI clock up to 52 MHz. Do you know the exact model of the memory chip? Can you test whether the 52 MHz frequency works on this device?